azkaban-common/src/main/java/azkaban/executor/ExecutionLogsDao.java

```diff
@@ -135,12 +135,25 @@ private void uploadLogFile(final DatabaseTransOperator transOperator, final int
     }
   }
 
-  int removeExecutionLogsByTime(final long millis)
+  int removeExecutionLogsByTime(final long millis, final int recordCleanupLimit)
+      throws ExecutorManagerException {
+    int totalRecordsRemoved = 0;
+    int removedRecords;
+    do {
+      removedRecords = removeExecutionLogsBatch(millis, recordCleanupLimit);
+      logger.debug("Removed batch of execution logs. Count of records removed in this batch: "
+          + removedRecords);
+      totalRecordsRemoved = totalRecordsRemoved + removedRecords;
```

Comment:
could we consider adding `sleep of 1` second here as recommended by the DBA?
Reference: http://mysql.rjweb.org/doc.php/deletebig

Reply:
Thank you. Adding it.